### PR TITLE
OEL-1294: Component scaffold.

### PR DESCRIPTION
**Pull request by donquixote**

### CODE REVIEW

**Comment by donquixote** [comment]
@brummbar as mentioned already in Jira, there are separate commits that we can individually discard if need be.

1. Review -> list of commits to be dropped.
2. Interactive rebase -> drop those commits, clean up.
3. Review again.

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    # Temporarily fix the triple store version to avoid "Provisional data" terms.
+    # @todo: Use the latest release once a solution for the "Provisional data" terms is implemented.
+    image: registry.fpfis.eu/openeuropa/triple-store-dev:1.14.0
+    pull: true
+    environment:
+      - SPARQL_UPDATE=true
+      - DBA_PASSWORD=dba
+
+pipeline:
+  composer-install-lowest:
```
Comment (brummbar) [inline .drone.yml]: Let's have this after install-highest, like it's in oe_starter_content. oe_content is kinda an anomaly, usually we have highest first.

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    group: prepare
+    image: registry.fpfis.eu/fpfis/httpd-php:${PHP_VERSION}-ci
+    volumes:
+      - /cache:/cache
+    commands:
+      - composer self-update --2
```
Comment (brummbar) [inline .drone.yml]: Let's take the chance to drop this, as QA reported it causes issues.

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      matrix:
+        COMPOSER_BOUNDARY: lowest
+
+  composer-install-highest:
+    group: prepare
+    image: registry.fpfis.eu/fpfis/httpd-php:${PHP_VERSION}-ci
+    volumes:
+      - /cache:/cache
+    commands:
+      - composer self-update --2
```
Comment (brummbar) [inline .drone.yml]: Same as above.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,141 @@
...
+* [Virtuoso 7 (Triplestore database)](https://github.com/openlink/virtuoso-opensource)
+
+#### Initial setup
+
+You can build the development site by running the following steps:
+
+* Install Virtuoso. See [OpenEuropa Content](https://github.com/openeuropa/oe_content/blob/3.x/README.md) for more information.
+
+* Customize build settings by copying `runner.yml.dist` to `runner.yml` and
+  changing relevant values, like your database credentials.
```
Comment (brummbar) [inline README.md]: Let's drop this, we don't support installing in any other way but docker.

File: runner.yml.dist [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  setup:phpunit:
+    - { task: "process", source: "phpunit.xml.dist", destination: "phpunit.xml" }
+  # Use the following commands only if you can run Task Runner commands on your host machine.
+  sparql:purge:
+    - "docker-compose exec sparql ./vendor/bin/robo purge"
+  sparql:import:
+    - "docker-compose exec sparql ./vendor/bin/robo import"
+  sparql:reset:
+    - { task: "run", command: "sparql:purge" }
+    - { task: "run", command: "sparql:import" }
```
Comment (brummbar) [inline runner.yml.dist]: Let's drop this section, it's not enough documented.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: .github_changelog_generator [hunk 6]
```diff
@@ -0,0 +1,2 @@
+release-branch=2.x
+exclude-tags-regex=^[0-1].[0-9]+.[0-9]+.*
```
Comment (brummbar) [inline .github_changelog_generator]: Actually this has to be removed, as it's not correct for our new component.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: .drone.yml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  web:
+    image: registry.fpfis.eu/fpfis/httpd-php:${PHP_VERSION}-ci
+    environment:
+      - DOCUMENT_ROOT=/test/oe_content_extra
+      - COMPOSERVER=--2
+  mysql:
+    image: registry.fpfis.eu/fpfis/sql:percona-5.7
+    command: --innodb-log-file-size=1G --max_allowed_packet=1G --innodb-buffer-pool-size=512M --wait_timeout=3000 --net_write_timeout=3000 --log_error_verbosity=3
+    environment:
+      - MYSQL_ALLOW_EMPTY_PASSWORD=yes
```
Comment (escuriola) [inline .drone.yml:25]: do we need MYSQL_DATABASE?
```suggestion
      - MYSQL_ALLOW_EMPTY_PASSWORD=yes
      - MYSQL_DATABASE=drupal
```

File: .drone.yml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+      - DBA_PASSWORD=dba
+
+pipeline:
+  composer-install-highest:
+    group: prepare
+    image: registry.fpfis.eu/fpfis/httpd-php:${PHP_VERSION}-ci
+    volumes:
+      - /cache:/cache
+    commands:
+      - composer install --ansi --no-progress
```
Comment (escuriola) [inline .drone.yml:42]:
```suggestion
      - composer self-update --2
      - composer install --ansi --no-progress
```

File: .drone.yml [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+      matrix:
+        COMPOSER_BOUNDARY: highest
+
+  composer-update-lowest:
+    group: prepare
+    image: registry.fpfis.eu/fpfis/httpd-php:${PHP_VERSION}-ci
+    volumes:
+      - /cache:/cache
+    commands:
+      - composer update --prefer-lowest --prefer-stable --ansi --no-progress
```
Comment (escuriola) [inline .drone.yml:53]:
```suggestion
      - composer self-update --2
      - composer update --prefer-lowest --prefer-stable --ansi --no-progress
```

**Review by escuriola — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 8 above)
Comment (brummbar) [reply .drone.yml:42]: We are dropping these ones, as we did in oe_showcase.

File: .drone.yml (hunk 9 above)
Comment (brummbar) [reply .drone.yml:53]: Same as above.

File: .drone.yml (hunk 7 above)
Comment (brummbar) [reply .drone.yml:25]: It's not really needed, Drupal will create the db anyway.

**Review by escuriola — APPROVED** [review]

**Review by brummbar — APPROVED** [review]

**Pull request merged**